modules/ding_wayf/ding_wayf.module

```diff
@@ -124,6 +124,81 @@ function ding_wayf_init() {
   }
 }
 
+
+/**
+ * Do a check on the checksum given from gatewayf.
+ *
+ * @param array $attributes
+ *   The attributes from gatewayf.
+ *
+ * @return bool
+ *   Safe or not.
+ */
+function ding_wayf_security_check(array $attributes) {
+  // Check attributes.
+  if (empty($attributes['schacPersonalUniqueID']) || empty($attributes['eduPersonTargetedID'])) {
+    // We expect alway to get these two attributes.
+    watchdog('ding_gatewayf', 'Neccesary attributes missing', array(), WATCHDOG_ERROR);
+    drupal_set_message('Sorry login failed - contact site administrator', 'warning');
+    return FALSE;
+  }
+
+  $gateway_checksum = isset($attributes['hash'][0]) ? $attributes['hash'][0] : NULL;
+  if (!isset($gateway_checksum)) {
+    // This is an error from gatewayf - checksum should always be in attributes.
+    watchdog('ding_gatewayf', 'No checksum from gatewayf', array(), WATCHDOG_ERROR);
+    return FALSE;
+  }
+
+  $fixed_params = array(
+    $attributes['schacPersonalUniqueID'][0],
+    $attributes['eduPersonTargetedID'][0],
+  );
+
+  $key = ding_wayf_key_from_attributes($fixed_params);
+  // Get salt from settings $conf.
+  $salt = variable_get('ding_gatewayf_salt', NULL);
```

Comment:
It seems odd that we get a variable prefixed with the name of a module (ding_gatewayf) which we probably are not using.